src/components/Questionnaire.js

```diff
@@ -253,50 +378,131 @@ query getQuestionnaire($slug: String!) {
           value
           category
         }
-        results: result {
+        choiceResults: result {
           count
           option {
             label
             value
             category
           }
         }
-        turnout {skipped submitted}
+      }
+      ... on QuestionTypeRange {
+        kind
+        ticks {
+          label
+          value
+        }
+        rangeResults: result {
+          histogram(ticks: $histogramTicks) {
+            x0
+            x1
+            count
+          }
+          mean
+          median
+          deviation
+        }
       }
     }
   }
 }
 `
 
 export default compose(
+  withApollo,
   withT,
   withMe,
   graphql(submitAnswerMutation, {
-    props: ({ mutate, ownProps: { slug } }) => ({
+    props: ({ mutate, ownProps: { histogramTicks } }) => ({
       submitAnswer: (question, payload, answerId) => {
         return mutate({
           variables: {
             answerId,
             questionId: question.id,
-            payload
+            payload,
+            histogramTicks
           },
           optimisticResponse: {
             __typename: 'Mutation',
             submitAnswer: {
-              __typename: 'QuestionTypeChoice',
-              id: question.id,
+              ...question,
+              userAnswer: {
+                __typename: 'Answer',
+                id: answerId,
+                payload
+              },
+
+              ...question.choiceResults && { choicesResults: question.choiceResults.map(r => ({
+                ...r,
+                count: r.count + (r.option.value == payload.value
+                  ? 1
+                  : 0
+                )
+              })) },
+
+              ...question.rangeResults && { rangeResults: {
+                ...question.rangeResults,
+                histogram: question.rangeResults.histogram.map(b => {
+                  if (payload.value >= b.x0 && payload.value < b.x1) {
+                    return { ...b, count: b.count + 1 }
+                  }
+
+                  return b
+                })
+              } },
+
+              turnout: {
+                ...question.turnout,
+                submitted: question.turnout.submitted + 1
+              }
+            }
+          }
+        })
+      }
+    })
+  }),
+  graphql(submitAnswerUnattributedMutation, {
+    props: ({ mutate, ownProps: { histogramTicks } }) => ({
+      submitAnswerUnattributed: (question, payload, answerId, pseudonym) => {
+        return mutate({
+          variables: {
+            answerId,
+            questionId: question.id,
+            payload,
+            pseudonym,
+            histogramTicks
+          },
+          optimisticResponse: {
```

Comment:
you could put the following code in a function and use it for `submitAnswert` and `submitAnswerUnattributed`